### PR TITLE
58 auth guard in frontend tetyana

### PR DESCRIPTION
**Pull request by TetyanaYahodska**

### CODE REVIEW

**Review by Gorbokonyk — DISMISSED** [review]

File: Frontend/src/app/app-routing.module.ts [hunk 1]
```diff
@@ -10,24 +10,27 @@ import { Error500Component } from "./errorpages/error500/error500.component";
 import { TwoFactorAuthenticationFormComponent } from "./two-factor-authentication-form/two-factor-authentication-form.component";
 import { UserSearchInterfaceComponent } from './user-search-interface/user-search-interface.component';
 import { SearchResponseComponent } from './search-response/search-response.component';
+import { UserAuthGuard } from './shared/user.auth.guard';
+import { AdminAuthGuard } from './shared/admin.auth.guard';
 
 const routes: Routes = [
   { path: '', redirectTo: '/login', pathMatch: 'full' },
   { path: 'admin/allSchemas', component: AdminManageSchemaListComponent },
```
Comment (YaroslavMalyarskyy) [inline Frontend/src/app/app-routing.module.ts:18]: user have access to admin page

File: Frontend/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -10,24 +10,27 @@ import { Error500Component } from "./errorpages/error500/error500.component";
 import { TwoFactorAuthenticationFormComponent } from "./two-factor-authentication-form/two-factor-authentication-form.component";
 import { UserSearchInterfaceComponent } from './user-search-interface/user-search-interface.component';
 import { SearchResponseComponent } from './search-response/search-response.component';
+import { UserAuthGuard } from './shared/user.auth.guard';
+import { AdminAuthGuard } from './shared/admin.auth.guard';
 
 const routes: Routes = [
   { path: '', redirectTo: '/login', pathMatch: 'full' },
```
Comment (YaroslavMalyarskyy) [inline Frontend/src/app/app-routing.module.ts:17]: the home page is always login even if we are logged in

File: Frontend/src/app/header/header.component.html [hunk 3]
```diff
@@ -5,6 +5,6 @@
     <span class="fs-4">CMIYC</span>
   </a>
   <div class="col-md-3 text-end float end">
-    <button type="button" class="btn btn-primary">Log out</button>
+    <button type="button" class="btn btn-primary" (click)="isLogout()">Log out</button>
```
Comment (YaroslavMalyarskyy) [inline Frontend/src/app/header/header.component.html:8]: log out does not work on the admin page

File: Frontend/src/app/shared/auth.service.ts [hunk 4]
```diff
@@ -24,4 +28,13 @@ export class AuthService {
     // we still need to handle the reception of the token
   }
 
+
+  logout(){
+    console.log('logout from authService');
+    localStorage.setItem('scope', 'null');
+    localStorage.setItem('access_token','null');
```
Comment (YaroslavMalyarskyy) [inline Frontend/src/app/shared/auth.service.ts:35]: does not work on the admin page, the token remains in the browser's memory

**Review by Gorbokonyk — APPROVED** [review]

**Review by YaroslavMalyarskyy — APPROVED** [review]

**Pull request merged**